newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -756,17 +752,22 @@ export default class AuthenticatedUserProvider extends React.Component<
           appLanguage: preferences.language,
           donateLink: form.donateLink,
           communityLinks: form.communityLinks,
+          survey: form.survey,
         }
       );
       await this._fetchUserProfileWithoutThrowingErrors();
       this.openEditProfileDialog(false);
     } catch (authError) {
       this.setState({ authError });
+      if (throwError) {
+        throw authError;
```

Comment:
why do you need this? Can't we just look at the authError in the state when it's finished to know if it went well or not?

Reply:
I think this `authError` is ill-named: it is a shared variable state where we store any error occurred during any API calls we do in the provider. This new `authenticationError` is truly an authentication error: an error that occurs only when we try to sign up or sign in an account.

Reply:
oh right, it does feel cumbersome having both though. And they both kindof have the same name, so it is slightly confusing